src/core/MaverickV2.sol

```diff
@@ -206,16 +232,35 @@ abstract contract MaverickV2 is SettlerAbstract {
         uint256 minBuyAmount
     ) internal returns (uint256 buyAmount) {
         bytes memory swapCallbackData = _encodeSwapCallback(permit, sig);
-        address pool = AddressDerivation.deriveDeterministicContract(maverickV2Factory, salt, maverickV2InitHash);
+        IMaverickV2Pool pool = IMaverickV2Pool(AddressDerivation.deriveDeterministicContract(maverickV2Factory, salt, maverickV2InitHash));
+        // Price P, given the tick and tick spacing is:
+        // (1) log_1.0001(P) = tick * tickSpacing
+        // To find the tick with a 100% price impact it is needed to find X such that:
+        // (2) log_1.0001(2 * P) = X * tickSpacing
+        // which is same as:
+        // (3) log_1.0001(2) + log_1.0001(P) = X * tickSpacing
+        // Substituting (1) into (3):
+        // (4) log_1.0001(2) + tick * tickSpacing = X * tickSpacing
+        // then it is possible to get to
+        // (5) log_1.0001(2) / tickSpacing = X - tick
+        // log_1.0001(2) is approximately 6931
```

Comment:
Make sure you're rounding in the direction you expect. Be explicit about your decision in a comment.